apps/drec-api/src/auth/is-drec-or-user.guard.ts

```diff
@@ -0,0 +1,24 @@
+import { ConfigService } from '@nestjs/config';
+import { CanActivate, ExecutionContext, Injectable } from '@nestjs/common';
+import { Request } from 'express';
+import { AuthGuard } from '@nestjs/passport';
+
+@Injectable()
+export class IsDrecOrUserGuard extends AuthGuard('jwt') implements CanActivate {
+  public constructor(private readonly configService: ConfigService) {
+    super({});
+  }
+
+  async canActivate(context: ExecutionContext): Promise<boolean> {
+    const request: Request = context.switchToHttp().getRequest();
+
+    if (
+      request.headers['api-key'] ===
+      this.configService.get<string>('DREC_API_KEY')
+    ) {
+      return true;
+    }
```

Comment:
Should this be replaced by the user with the Admin role?

Or do we also use API keys in D-REC?

Reply:
Is D-REC pushing data to our API like SBE is? In that cause the API key should stay, otherwise we can just check IsRole(Admin)

Reply:
I was thinking of removing it.. I don`t have any info about needing API keys.
Should I go ahead and remove it?

Reply:
For now, we don`t know yet. I will double check with Jonathan and if it`s not needed, I will remove it.

Reply:
Just checked and they won`t be pushing data, we will build a module to pull data from an API and push it in our system.

Reply:
Ok, then we can should remove the API key entirely.

Reply:
Done